src/main-process/file-recovery-service.js

```diff
@@ -146,6 +147,7 @@ async function tryStatFile (path) {
 }
 
 async function copyFile (source, destination, mode) {
+  await mkdirp(Path.dirname(destination))
```

Comment:
Does this work? From the `mkdirp` docs, I didn't think it returned a promise; I thought you had to pass a callback.

Reply:
Can someone help me rework the code? I am having trouble understanding how to use the callback and return a promise.